python/tvm/container.py

```diff
@@ -17,9 +17,15 @@ def __getitem__(self, i):
             start = i.start if i.start is not None else 0
             stop = i.stop if i.stop is not None else len(self)
             step = i.step if i.step is not None else 1
+            if start < 0:
+                start += len(self)
+            if stop < 0:
+                stop += len(self)
             return [self[idx] for idx in range(start, stop, step)]
 
-        if i >= len(self):
+        if i < 0:
+            i += len(self)
+        if i < 0 or i >= len(self):
```

Comment:
would you mind also print original `i` value and array size in the IndexError.

Reply:
i can print original `i` and array size here, but this is difficult for slice because it calls `__getitem__` with transformed index